### DISCUSSION TITLE
Using MediaLibrary Uploaders with Relationship field

### DISCUSSION THREAD

**Discussion by prodixx** [Laravel-Backpack/community-forum#979] [Q&A (Help)]
so, i'm trying to create a relationship field with media library uploader as subfield.

I have 2 Models : Product and AttributeOption with pivot table. Here is my setup:
```php
// Product Model implements HasMedia and uses InteractsWithMedia
public function attributesOptions()
{
    return $this->belongsToMany(\App\Models\AttributeOption::class, 'attribute_option_product')
        ->withPivot('image')
        ->using(\App\Models\ProductAttributeOption::class)
        ->withTimestamps();
}

// AttributeOption Model also implements HasMedia and uses InteractsWithMedia and has the same relationship

// ProductAttributeOption Model
class ProductAttributeOption extends Pivot implements HasMedia
{
    use InteractsWithMedia;

    protected $table = 'attribute_option_product';
}

// ProductCrudController
[
    'name' => 'glassAttributes',
    'label' => 'Glass',
    'type' => 'relationship',
    'tab' => 'Attribute',
    'subfields' => [
        [
            'name' => 'image',
            'type' => 'upload',
            'label' => 'Image',
            'withMedia' => [],
            'wrapper' => ['class' => 'form-group col-md-2'],
        ],
    ],
    'pivotSelect' => [
        'label' => 'Glass',
        'placeholder' => 'select',
        'wrapper' => ['class' => 'col-md-4'],
        'options' => (function ($query) {
            return $query->onlyGlassCategory()->get();
        }),
    ],
    'new_item_label' => 'Add glass',
    'init_rows' => 1,
    'min_rows' => 0,
    'max_rows' => 3,
    'reorder' => false,
]
```

### What happened
im getting this error 
```
Method Spatie\MediaLibrary\MediaCollections\Models\Collections\MediaCollection::delete does not exist.
```

If i remove media library setup from pivot model, i get this error:
```
Call to undefined method App\Models\ProductAttributeOption::getMedia()
```

Any idea how to make this work?
Thanks

**Comment by karandatwani92** [comment]
Hey @pxpm 

- I did it with `withFiles`, and this works fine -> check [gist](https://gist.github.com/karandatwani92/953e5817692529f1cb54514f7b2b5d18)
- When I change to `withMedia`, it throws https://flareapp.io/share/q5YJKODm
What do you suggest?

https://github.com/Laravel-Backpack/community-forum/assets/8214221/e5277e84-5ebf-4fc0-9923-4a6a8bd6b60c

**Comment by pxpm** [comment]
Hey @prodixx hope to find you well.
This seems like a bug on our side, I am guessing we probably wanted to call `->delete()` on a model, but we are calling `->delete()` on a media collection. 

If you investigate this and come up with a fix, please go ahead and send a PR, I will review it and merge it if it's a quick win. 

We are doing a big re-work on uploaders, adding tests and all that stuff, and media-library will need to be updated too after that. It's in https://github.com/Laravel-Backpack/CRUD/pull/5478

Cheers
